src/mptool.f90

```diff
@@ -143,5 +290,343 @@ pure function cross_product(a,b)
     cross_product(3) = a(1) * b(2) - a(2) * b(1)
     !
   end function cross_product
+
+  subroutine pwrite_1darray(filename,data_1,data_2,data_3,data_4,data_5,data_6)
+
+    ! arguments
+    character(len=*),intent(in) :: filename
+    real(mytype),intent(in) :: data_1(:)
+    real(mytype),intent(in),optional :: data_2(:),data_3(:),data_4(:),data_5(:),data_6(:)
+
+    ! local data
+    integer :: local_size,total_size
+    integer :: ierr, fh, datatype, status(mpi_status_size)
+    integer(kind=mpi_offset_kind) :: offset
+
+    ! get the size of the array
+    local_size=size(data_1)
+
+    ! calculate the offset for each process
+    offset = prelay(local_size)*mytype
```

Comment:
Are we sure this statement based on the value of `mytype` is portable ?

Reply:
I tested with single and double precision, and it works fine. But I have replaced it here.